libvirt/domain.go

```diff
@@ -280,17 +281,115 @@ func inlineUpdateCoreOsIgnition(d *schema.ResourceData, domainDef *libvirtxml.Do
 		if err != nil {
 			return err
 		}
-		err = guestfishExecution(rootPath, fileToUpload)
+		connURI, err := virConn.GetURI()
+		if err != nil {
+			return err
+		}
+		err = guestfishExecution(rootPath, fileToUpload, connURI)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func getLibvirtHostFromURI(connURI string) string {
+	// Refer to libvirt URI: https://libvirt.org/uri.html
+	virHost := strings.Split(strings.SplitN(connURI, "//", 2)[1], "/")[0]
+	if strings.Contains(virHost, "@") {
+		virHost = strings.SplitN(virHost, "@", 2)[1]
```

Comment:
Coincidently the `username@` URL is mostly likely used with SSH libvirt URI so do we really want to ignore username in the URL?

Reply:
In this change, username defaults to `root`, since normally username should be `root` in a libvirt URI, as libvirtd run as user `root`. And the ssh remote call need write access to modify the remote libvirt volume.

Reply:
Not at all. system libvirt demon runs as root but you don't have to be root to connect to it. There is hope we (Openshift) will move away from system demon in the future. Allowing remote root access is not the best idea anyway. So no, this assumption is not correct IMO.